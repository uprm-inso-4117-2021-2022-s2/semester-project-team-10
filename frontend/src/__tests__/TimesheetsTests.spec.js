import timesheets from './timesheets.json'
import '../services/auth.service.js'

test('the timesheet data is correct', () => {
    expect(timesheets).toMatchSnapshot();
    });

    for (let i = 0; i < timesheets.length; i += 1){
        test(`timesheets[${i}] should have properties (employee_id, end_time, start_time, timesheet_id, work_desc)`, () => {
            expect(timesheets[i]).toHaveProperty('employee_id');
            expect(timesheets[i]).toHaveProperty('end_time');
            expect(timesheets[i]).toHaveProperty('start_time');
            expect(timesheets[i]).toHaveProperty('timesheet_id');
            expect(timesheets[i]).toHaveProperty('work_desc');
        });
};

test('spying (original implementation) an axios GET request for timesheet', () => {
    const timesheet = {
        employee_id: n => `Employee ID: ${n}`
    };
    const spy = jest.spyOn(timesheet, 'employee_id');
    expect(timesheet.employee_id(1)).toBe('Employee ID: 1');
    expect(spy).toHaveBeenCalledWith(1);
});
