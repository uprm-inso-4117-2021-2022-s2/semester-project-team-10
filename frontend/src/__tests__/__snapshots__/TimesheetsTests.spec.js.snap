// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the timesheet data is correct 1`] = `
Array [
  Object {
    "employee_id": 1,
    "end_time": "Thu, 18 Mar 2022 18:15:00 GMT",
    "start_time": "Thu, 18 Mar 2022 17:30:00 GMT",
    "timesheet_id": 1,
    "work_desc": "Some description",
  },
  Object {
    "employee_id": 2,
    "end_time": "Thu, 18 Mar 2022 12:15:00 GMT",
    "start_time": "Thu, 18 Mar 2022 11:30:00 GMT",
    "timesheet_id": 2,
    "work_desc": "UI/UX research",
  },
  Object {
    "employee_id": 3,
    "end_time": "Thu, 18 Mar 2022 9:15:00 GMT",
    "start_time": "Thu, 18 Mar 2022 8:30:00 GMT",
    "timesheet_id": 3,
    "work_desc": "Security patrol",
  },
  Object {
    "employee_id": 4,
    "end_time": "Thu, 18 Mar 2022 17:15:00 GMT",
    "start_time": "Thu, 18 Mar 2022 18:30:00 GMT",
    "timesheet_id": 4,
    "work_desc": "UI/UX research",
  },
  Object {
    "employee_id": 5,
    "end_time": "Thu, 18 Mar 2022 17:15:00 GMT",
    "start_time": "Thu, 18 Mar 2022 18:30:00 GMT",
    "timesheet_id": 5,
    "work_desc": "SEO research",
  },
]
`;
